### DISCUSSION TITLE
Blurred/Fade to Black Border on Images

### DISCUSSION THREAD

**Discussion by DAConsulting** [ImageMagick/ImageMagick#2759] [Help]
Are there any examples of command lines to accomplish  putting a blurred/fade-to-black border on a collection of images?

I see information about Gaussian blurs, but I'm not clear how to only apply this to the border, progressively toward the edges.

Searches don't turn up much.

All leads appreciated.

I've attached a "before" and "after" image to give an idea what I'm after.

Thanks,

- Dave
![Photo Jan 05, 11 16 37 AM](https://user-images.githubusercontent.com/49458908/97038175-6615f900-1538-11eb-927a-995334a75fef.jpg)

![Photo Jan 05, 11 16 37 AM-feathered](https://user-images.githubusercontent.com/49458908/97038206-6d3d0700-1538-11eb-920c-95b1537c3c00.jpg)

**Comment by fmw42** [comment]
What platform?  What version of ImageMagick.  You can create a rounded rectangular mask and use that to blend in to black.

One way to do that would be in IM 7 Unix syntax:

```
magick bass.jpg -auto-orient \
-set option:wd "%[fx:0.45*w]" -set option:ht "%[fx:0.45*h]" -set option:center "%[fx:w/2],%[fx:h/2]" \
\( -clone 0 -fill black -colorize 100 -fill white \
-draw "translate %[center] roundrectangle -%[wd],-%[ht] %[wd],%[ht] 50,50" \
-resize 50% -blur 0x50 -resize 200% \) \
-compose multiply -composite bass_vignette

```

![bass_vignette](https://user-images.githubusercontent.com/13458218/97042944-a53b4e80-1526-11eb-8637-8a259f55a309.jpg)



An alternate simle method, produces an oval shape:

`magick bass.jpg -auto-orient -background black -vignette 0x50+2+2% bass_vignette2.jpg`


![bass_vignette2](https://user-images.githubusercontent.com/13458218/97043640-a15bfc00-1527-11eb-8fb4-5f461a7ef494.jpg)

**Comment by GeeMack** [comment]
Here is a method using ImageMagick v6 for applying a blurred border to an input image...

    convert inimage.png -write mpr:img0 -evaluate set 100% \
       -bordercolor black -shave 200 -border 200 -blur 0x100 \
       mpr:img0 -compose multiply -composite outimage.png

That starts by reading the input and storing a copy of it in a temporary memory register named "mpr:img0". Then it sets all the color values to 100% to change the entire image to white.

Next set the border color to black, shave 200 pixels from every side, and add the 200 pixels back as a black border. Blur the resulting white with black border image to create a mask. This makes a good size blurred border for the 3024x4032 sample image you posted. Obviously you'd use the amount of shaving, bordering, and blurring necessary for your task. 

Then bring the memory copy of the original image "mpr:img0" back into the command. Set the compose method to "multiply" and composite the input image onto that grayscale blurred mask to create the output.

This works for me with IMv6.8 in a bash shell. It works with IMv6.9 in Windows by changing the continued-line backslashes "\\" to carets "^". It should work just as well with IMv7 by changing the command "convert" to "magick".

Edited to add:

I added `-resize 20%` before the output of the above command to make a smaller example of the result.

![outimage](https://user-images.githubusercontent.com/65475279/97066647-954e5980-157c-11eb-854f-1be65757b9fd.png)
